src/app/test/router.ts

```diff
@@ -11,12 +11,16 @@ const RunTestsSchema = validateRequest({
 export function testRouter() {
   const router = express.Router()
 
-  router.post('/run-tests', RunTestsSchema, async (req, res) => {
+  router.post('/run-tests', RunTestsSchema, async (req, res, next) => {
     const testSession = createTestSession(req.body)
+    try {
+      const result = await runTests(testSession)
 
-    const result = await runTests(testSession)
-
-    return res.json(result.toTestResponse())
+      return res.json(result.toTestResponse())
+    } catch (e) {
+      finalizeTestSession(testSession)
+      next(e)
+    }
   })
```

Comment:
It's close, but now if the `createTestSession` will throw, the error won't be propagated to the error handler and it will crash the server.

Reply:
Done 👍